pkg/domain/entities/types/containers.go

```diff
@@ -37,6 +37,9 @@ type ContainerStatsReport struct {
 
 type ContainerUpdateOptions struct {
 	NameOrID                        string
-	Specgen                         *specgen.SpecGenerator
+	Resources                       *specs.LinuxResources
+	DevicesLimits                   *define.UpdateContainerDevicesLimits
 	ChangedHealthCheckConfiguration *define.UpdateHealthCheckConfig
+	RestartPolicy                   *string
+	RestartRetries                  *uint
```

Comment:
This type is used in pkg/bindings/containers Update() function as such we cannot chnage the type as this will break bindings users, we can add new fields but we cannot remove the function of old fields.